core/spec/models/spree/order_inventory_spec.rb

```diff
@@ -189,10 +193,12 @@
       end
 
       it 'only attempts to destroy as many units as are eligible, and return amount destroyed' do
-        allow(shipment).to receive_messages(inventory_units_for_item: [
-          mock_model(Spree::InventoryUnit, :variant_id => variant.id, :state => 'shipped'),
-          mock_model(Spree::InventoryUnit, :variant_id => variant.id, :state => 'on_hand')
-        ])
+        allow(shipment).to receive_messages(
+          inventory_units_for_item: [
+            mock_model(Spree::InventoryUnit, variant_id: variant.id, state: 'shipped', shipped?: false),
```

Comment:
@priyank-gupta Why have you added ```shipped?: false``` to these specs?
If ```state: 'shipped'```, isn't natural for ```shipped?``` to return true?. What am I missing here?